framework/src/modules/chain/blocks/fork_choice_rule.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Determine if Case 2 fulfills
+ * @param lastBlock
+ * @param currentBlock
+ * @return {boolean}
+ * @private
+ */
+// eslint-disable-next-line class-methods-use-this
+const isValidBlock = (lastBlock, currentBlock) =>
+	lastBlock.height + 1 === currentBlock.height &&
+	lastBlock.id === currentBlock.previousBlock;
+
+/**
+ * Determine if Case 1 fulfills
+ * @param lastBlock
+ * @param currentBlock
+ * @return {boolean}
+ * @private
+ */
+// eslint-disable-next-line class-methods-use-this
+const isIdenticalBlock = (lastBlock, currentBlock) =>
+	lastBlock.id === currentBlock.id;
+
+/**
+ * Determine if two blocks are duplicates
+ * @param lastBlock
+ * @param currentBlock
+ * @return {boolean}
+ * @private
+ */
+// eslint-disable-next-line class-methods-use-this
+const isDuplicateBlock = (lastBlock, currentBlock) =>
+	lastBlock.height === currentBlock.height &&
+	lastBlock.prevotedConfirmedUptoHeight ===
+		currentBlock.prevotedConfirmedUptoHeight &&
+	lastBlock.previousBlock === currentBlock.previousBlock;
+
+/**
+ * Determine if Case 3 fulfills
+ * @param lastBlock
+ * @param currentBlock
+ * @return {*|boolean}
```

Comment:
Is this not just a boolean being returned? (Also for the other functions)

Reply:
Solved